### PR TITLE
Handle None weights in rotation_matrix correctly

### PR DESCRIPTION
**Pull request by kain88-de**
QCPROT expects a valid array and not a value of Type None. Using
`rotation_matrix` directly did raise a type exception because of this.
Now rotation_matrix generates equal weights as input if none are given
and the doc-strings in QCPROT are adjusted to reflect the function
signatures.

### CODE REVIEW

File: package/MDAnalysis/lib/qcprot.pyx [hunk 1]
```diff
@@ -141,33 +141,37 @@ def InnerProduct(np.ndarray[np.float64_t,ndim=1] A,
...
+    ----------
+    A : ndarray float64_t
+        result inner product array, modified in place
+    coords1 : ndarray float64_t
+        reference structure
+    coord2 : ndarray float64_t
+        candidate structure
+    N : int
+        size of system
+    weights : ndarray float64_t
```
Comment (kain88-de) [inline package/MDAnalysis/lib/qcprot.pyx]: does anyone know how to properly annotate the type in numpy-style doc strings?

Comment (orbeckst) [inline package/MDAnalysis/lib/qcprot.pyx]: Not really, but the type would be `numpy.float64`.

**Comment by orbeckst** [comment]
I don't understand why you changed the code. It was able to handle `weights=None` previously and it took a (probably) marginally faster code path that leaves out the multiplications with the weights.

Was the idea to make the code more readable?

Did you benchmark before/after your changes?

**Comment by kain88-de** [comment]
The code is already typed. It couldn't be given a `None` for the weights. I've run into this today when I tried to use the `matrix_rotation` function. So I also removed the code paths that weren't usable.

**Comment by kain88-de** [comment]
OK I have to check a little better why I'm getting a TypeError currently

**Comment by orbeckst** [comment]
On 8 Feb, 2016, at 14:06, kain88-de wrote:

> The code is already typed. It couldn't be given a None for the weights. I've run into this today when I tried to use the matrix_rotation function. So I also removed the code paths that weren't usable.

Then I have no idea why the tests passed that used an explicit `None` for weights! See https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/test_qcprot.py#L109

**Comment by kain88-de** [comment]
The error is a different one. I opened a issue for it

**Pull request closed without merging**